client/src/pages/Register/PaymentStep.tsx

```diff
@@ -98,68 +158,89 @@ function PaymentStep(props: RegisterStepProps) {
+    payPalOptions['client-id'] = 'sb';
+  }
+
   return (
-    <div className="payment-form">
-      <h1>Choose your payment option</h1>
-      <p>
-        If you need to go back and change a value, refresh this page and it
-        will take you back to the registration form with your values filled
-        in
-      </p>
-      <h3>Total: ${paymentData.total}</h3>
-      {
-        props.deposit && (
-          <JsonSchemaForm
-            schema={{ 
-              type: 'object',
-              properties: {
+    <PageWrapper>
+      <div className="payment-form" style={loading ? { pointerEvents: 'none' } : undefined}>
+        {
+          !!loading && (
+            <div className="payment-disable-overlay">
+              <div>
+                <Spinner />
+              </div>
+            </div>
+          )
+        }
+        <h1>Choose your payment option</h1>
+        <h3>Total: ${(total || 0).toFixed(2)}</h3>
+        {
+          paymentStep.deposit && (
+            <JsonSchemaForm
+              schema={{ 
+                type: 'object',
+                properties: {
+                  deposit: {
+                    type: 'string',
+                    ...paymentStep.deposit,
+                  }
+                }
+              }}
+              uiSchema={{
                 deposit: {
-                  type: 'string',
-                  ...props.deposit,
+                  'ui:placeholder': 'Choose an option',
                 }
-              }
-            }}
-            uiSchema={{
-              deposit: {
-                'ui:placeholder': 'Choose an option',
-              }
-            }}
-            onChange={(arg) => setDepositChoice(arg.formData)}
-            formData={depositChoice}
-            templateData={{ }}
-            noHtml5Validate
-          >&nbsp;</JsonSchemaForm>
-        )
-      }
-      {
-        !!loading && (
-          <div className="payment-disable-overlay">
-            <div>
-              <Spinner animation="border" />
-            </div>
-          </div>
-        )
-      }
-      <button
-        onClick={submitPayByCheck}
-        className="payby-check-button"
-      >
-        <img alt="Check" src={checkImage} />
-        Check
-      </button>
-      <PayPalButtons
-        {...props}
-        setLoading={setLoading}
-        payPalCreateOrder={payPalCreateOrder}
-        payPalOnApprove={payPalOnApprove}
-      />
-    </div>
+              }}
+              onChange={onDepositChange}
+              formData={depositChoice}
+              templateData={{ }}
+              noHtml5Validate
+            >&nbsp;</JsonSchemaForm>
+          )
+        }
+        <button
+          onClick={submitPayByCheck}
+          className="payby-check-button"
+        >
+          <img alt="Check" src={checkImage} />
+          Check
+        </button>
+        {
+          !!payPalOptions && (
```

Comment:
I had to move the PayPal top level provider to this component to make the above listener work.